Sources/NIO/ByteBuffer-aux.swift

```diff
@@ -373,6 +373,39 @@ extension ByteBuffer {
         self._moveWriterIndex(forwardBy: written)
         return written
     }
+    
+    /// Writes `byte` `count` times. Moves the writer index forward by the number of bytes written.
+    ///
+    /// - parameter byte: The `UInt8` byte to repeat.
+    /// - parameter count: How many times to repeat the given `byte`
+    /// - returns: How many bytes were written.
+    @discardableResult
+    public mutating func writeRepeatingByte(_ byte: UInt8, count: Int) -> Int {
+        precondition(count >= 0, "Can't write fewer than 0 bytes")
+        let written = self.setRepeatingByte(byte, count: count, at: self.writerIndex)
+        self._moveWriterIndex(forwardBy: written)
+        return written
+    }
+    
+    /// Sets the given `byte` `count` times at the given `index`. Will reserve more memory if necessary. Does not move the writer index.
+    ///
+    /// - parameter byte: The `UInt8` byte to repeat.
+    /// - parameter count: How many times to repeat the given `byte`
+    /// - returns: How many bytes were written.
+    @discardableResult
+    public mutating func setRepeatingByte(_ byte: UInt8, count: Int, at index: Int) -> Int {
+        precondition(count >= 0, "Can't write fewer than 0 bytes")
+        let bytesCount = count
+        let newEndIndex: _Index = UInt32(index) + _toIndex(bytesCount)
+        if !isKnownUniquelyReferenced(&self._storage) {
```

Comment:
- We shouldn't do the CoW-thing manually here yet, this is too dangerous.
- Please use

```swift
self.reserveCapacity(index + count)
self.withVeryUnsafeMutableBytes { destCompleteStorage in
            assert(destCompleteStorage.count >= index + count)
            let dest = destCompleteStorage[index ..< index + count]
            memset(dest.baseAddress!, byte, count: count)
        }
        return count
}
```